### ISSUE TITLE
Reusing ResGCN

### ISSUE THREAD

**Issue by MritulaC** [yfsong0709/ResGCNv1#9]
Hi,
 I want to use ResGCN for my human walking time-series data analysis.  How am I to do that? Where should I start?

Thanks
Mritula

**Comment by yfsong0709** [comment]
ResGCN is a method to handle skeleton data. If you want to utilize it to your task, please make sure that your data is formed by 3D skeleton sequences. Then, resize your data shape to (N, C, T, V, M), where N denotes batch size, C=3 denotes 3D coordinates, T denotes the frame number in a sequence, V denotes the joint number in a skeleton, and M denotes the skeleton number in a frame. After that, modify the dataloader (see 'src/dataset/*' for more help). Finally, the model is able to be trained.

Hope this comment is helpful to you.

**Comment by MritulaC** [comment]
Hi,
 Thanks for your reply,
 Yes, I have 3D skeleton keypoints (14 Keypoints) per frame. 

 1.How am I modify the following with respect to my dataset?
    "python scripts/modify_configs.py --path <path/to/save/preprocessed/data> --ntu60_path <path/to/ntu60/dataset>"
As I have only one dataset,

2. Also, what format should my Input data be?

Thanks,
 Mritula

**Comment by rd20karim** [comment]
@yfsong0709 @MritulaC
I work In the same problem and want to know the format and numpy structure on which we should save our skeleton data with corresponding action labels, and for the skeleton graph what's is the minimum changes should we do and where(specify connected joints,...)
Thank's.

**Comment by Anirudh257** [comment]
@rd20karim save the file in the numpy structure of skeleton data, i.e. (N x C x T x V x M). Make the changes in the graph structure given in the **src/graph/nets.py** and some changes would be required in other config files as well.

**Comment by rd20karim** [comment]
Thank's for your clarifications , but always I ask how to specify the target which represent the **class label for the action recognition task** in the input data ?, 
after this the next question is where to save this data ? I try to find a rapid way without the need to read all the project but maybe it's not possible

**Comment by Anirudh257** [comment]
@rd20karim I am sorry if I didn't understand your question properly earlier. The target variable **y** given in the **src/processor.py** is a number in the range **[0, num_classes]**, where **num_classes** represents the number of classes in your dataset. It is not present within the **input data ** but is stored independently from it in a **pickle file** that can be created automatically by following the steps to prepare the data as given in the code: https://github.com/yfsong0709/ResGCNv1/blob/master/scripts/auto_gen_data.sh.

My directory structure looks like this for ResGCN:

```
--main.py
--data/ntu120/
...
--data/ntu/train_label.pkl
--data/ntu/test_label.pkl
```

If you still have doubts, feel free to reach out on my mail id: at794@snu.edu.in

**Comment by rd20karim** [comment]
Thank you so much for your reply's , I will check for the rest procedure to customize the code
